marketplaceItems/avatarCustomization101/appResources/appData/resources/modules/mirrorClient.js

```diff
@@ -0,0 +1,150 @@
+    // overlay's position. Deletes and re-adds the mirror overlay so the url and position is updated.
+    function updateMirrorOverlay() {
+        if (mirrorOverlayRunning) {
+            var mirrorProps = Entities.getEntityProperties(_this.entityID, ["rotation", "dimensions", "position"]);
+            var dimX = mirrorProps.dimensions.x;
+            var dimY = mirrorProps.dimensions.y;
+            
+            Overlays.deleteOverlay(mirrorOverlayID);
+            mirrorOverlayID = Overlays.addOverlay("image3d", {
+                name: "mirrorOverlay",
+                url: "resource://spectatorCameraFrame",
+                emissive: true,
+                parentID: _this.entityID,
+                alpha: 1,
+                localRotation: ZERO_ROT,
+                localPosition: { 
+                    x: 0,
+                    y: 0,
+                    z: mirrorOverlayOffset
+                }
+            });
+            updateMirrorDimensions(true);
+        }
+    }
+
+    _this.calculateMirrorResolution = function(entityDimensions) {
+        var mirrorResolutionX, mirrorResolutionY;
+        if (entityDimensions.x > entityDimensions.y) {
+            mirrorResolutionX = MAX_MIRROR_RESOLUTION_SIDE_PX;
+            mirrorResolutionY = Math.round(mirrorResolutionX * entityDimensions.y / entityDimensions.x);
+        } else {
+            mirrorResolutionY = MAX_MIRROR_RESOLUTION_SIDE_PX;
+            mirrorResolutionX = Math.round(mirrorResolutionY * entityDimensions.x / entityDimensions.y);
+        }
+
+        var resolution = {
+            "x": mirrorResolutionX,
+            "y": mirrorResolutionY
+        };
+
+        return resolution;
+    };
+    
+    // Sets up spectator camera to render the mirror, calls 'updateMirrorOverlay' once to set up
+    // mirror overlay, then connects 'updateMirrorDimensions' to update dimension changes
+    _this.mirrorOverlayOn = function(onPreload) {
+        if (!mirrorOverlayRunning) {
+            if (!spectatorCameraConfig.attachedEntityId) {
+                mirrorOverlayRunning = true;
+                spectatorCameraConfig.mirrorProjection = true;
+                spectatorCameraConfig.attachedEntityId = _this.entityID;
+                previousFarClipDistance = spectatorCameraConfig.farClipPlaneDistance;
+                spectatorCameraConfig.farClipPlaneDistance = FAR_CLIP_DISTANCE;
+                Render.getConfig("SecondaryCameraJob.ToneMapping").curve = 0;
+                var mirrorEntityDimensions = Entities.getEntityProperties(_this.entityID, 'dimensions').dimensions;
+                var initialResolution = _this.calculateMirrorResolution(mirrorEntityDimensions);
+                spectatorCameraConfig.resetSizeSpectatorCamera(initialResolution.x, initialResolution.y);
+                spectatorCameraConfig.enableSecondaryCameraRenderConfigs(true);
+                updateMirrorOverlay();
+                Script.update.connect(updateMirrorDimensions);
```

Comment:
does this need to be connected? why does `updateMirrorDimensions` need to be called so often?

Reply:
I likely need help with understanding this one.